configsuite/extension/ext.py

```diff
@@ -0,0 +1,206 @@
+        classes=["cs_children"], ids=["cs_{}_children".format(title)]
+    )
+
+    if schema[MK.Type] == types.NamedDict:
+        example = dict()
+        for key, value in schema[MK.Content].items():
+            generate(value, section_children, key)
+
+        for child in section_children.children:
+            example[child["parent"]] = child["example"]
+
+    elif schema[MK.Type] == types.List:
+        generate(
+            schema[MK.Content][MK.Item], section_children, parent_title="List Item"
+        )
+        example = [section_children.children[0]["example"]]
+
+    elif schema[MK.Type] == types.Dict:
+        example = dict()
+        generate(schema[MK.Content][MK.Key], section_children, parent_title="Key")
+        generate(schema[MK.Content][MK.Value], section_children, parent_title="Value")
+        example[section_children.children[0]["example"]] = section_children.children[1][
+            "example"
+        ]
+
+    elif isinstance(schema[MK.Type], types.BasicType):
+        example = "{}".format(BASIC_EXAMPLE.get(schema[MK.Type].name, "undefined"))
+        section_children += nodes.paragraph(
+            text=":type: {_type}".format(_type=schema[MK.Type].name)
+        )
+
+    else:
+        err_msg = "Unexpected type ({}) in schema while generating documentation."
+        raise TypeError(err_msg.format(schema[MK.Type]))
+
+    section_content += section_children
+
+    section += section_content
+    section["example"] = example
+    section["parent"] = parent_title
+
+    parent_section += section
+
+
+class ConfigsuiteDocDirective(Directive):
+    has_content = True
+    option_spec = {"module": unchanged_required}
+
+    def run(self):
+        if "module" not in self.options:
+            raise self.error("Required :module: argument missing")
+
+        schema_function_string = self.options.get("module")
+
+        if "." not in schema_function_string:
+            raise self.error(
+                "Configsuite :module: has invalid format. "
+                + "Expected 'module.function', got {}".format(schema_function_string)
+            )
+
+        module_str, func_str = schema_function_string.rsplit(".", 1)
+
+        try:
+            module = importlib.import_module(module_str)
+        except (ImportError) as e:
+            raise self.error(str(e))
+
+        try:
+            schema = getattr(module, func_str)()
+        except (AttributeError) as e:
+            raise self.error(str(e))
+
+        section = cs_container(classes=["cs_top_container"])
+        generate(schema, section)
+        return [section]
+
+
+def setup(app):
+    # Add our static path
+    app.connect("builder-inited", scb_static_path)
+
+    # Add relevant code to headers
+    app.add_css_file("configsuite_ext.css")
+    app.add_js_file("configsuite_ext.js")
+
+    app.add_node(
+        cs_container,
+        html=(visit_container, depart_container),
+        latex=(visit_container, depart_container),
```

Comment:
the latex and text can't have the same nodes as html. We either have to supply individual nodes for all builders, or only support html.

Reply:
This PR will only handle `html`, the others have been removed